Clarify "transition matrix" in docs (#2144)
tskit-dev/msprime: docs/mutations.md

### Matrix Mutation Models

These classes are defined by an alphabet of possible alleles (`alleles`); an array of
probabilities that determines how likely each allele is to be the root or ancestral

> allele (`root_distribution`); and a `transition_matrix` specifying the probabilities,
> given a mutation has occurred, that each allele will mutate to every other allele.

Reviewer: ```suggestion
allele (`root_distribution`); and a `transition_matrix` specifying the probabilities
with which a given mutation changes each allele into each other allele.
```

Author: Nice wording.

divided evenly between 💩 and 🔥. Here is the resulting "genotype matrix":

```{code-cell}

for v in mts.variants():
   print("".join(v.alleles[k] for k in v.genotypes))

```

(sec_mutations_matrix_mutation_theory)=

#### Parameterisation of Matrix Mutation Models

Mutation matrix models are specified by three things: an alphabet,
a root distribution, and a transition matrix.
These leave one free parameter: an overall mutation rate,
specified by the mutation `rate` in the call to {func}`.sim_mutations`.
Concisely,
the underlying model of mutation is a continuous-time Markov chain on the alphabet,
started by a draw from `root_distribution`, and
with instantaneous transition rate from `i` to `j` that is equal to
`rate` multiplied by `transition_matrix[i,j]`.
The `root distribution` and every row in the `transition_matrix`

> must give *probabilities*, i.e., they must be nonnegative numbers summing to 1.

Reviewer: good catch!

(probably) contain silent mutations.

Some facts about Markov chains are useful to interpret the statistics
of these mutation models. First, suppose we have tabulated all mutations, and so
for each pair of alleles {math}`i` and {math}`j` we have the proportion of mutations
that caused an {math}`i \to j` change. If allele {math}`i` mutates to a different allele,
the chance it mutates to allele {math}`j` is proportional to `transition_matrix[i,j]` but
excluding the diagonal (no-change) entry, so is equal to
`transition_matrix[i,j] / (1 - transition_matrix[i,i])`.
Second, suppose that an ancestor carries allele {math}`i` at a given position.

> The probability {math}`P(t)` that her descendant some time {math}`t` in the future

Reviewer: We shouldn't introduce new notation (here, P(t)) if we don't actually use the notation anywhere. Perhaps add the equation (in math) that P(t) = e^{tQ}?

Reviewer: whoops that's P(t) = e^{t r Q}, where r is the rate

carries allele {math}`j` can be found using the scaled
[infinitesimal rate matrix](<https://en.wikipedia.org/wiki/Transition_rate_matrix>) of
the Markov chain, {math}`Q`, and taking its matrix exponential:

```{code-cell} python
import scipy

Q = model.transition_matrix - np.eye(len(model.alleles))
t = 0.1  # time
scaled_Q = t * rate * Q
Pt = scipy.linalg.expm(scaled_Q)

print("P(t) = probs of allele change along a branch of length", t)
print()

> print("    ----  End  state ----")

Reviewer: I'm not sure what "end state" means here?

Reviewer: oh wait got it, nm
